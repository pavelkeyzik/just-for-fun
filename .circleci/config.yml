version: 2
jobs:
  build:
    working_directory: ~/just-for-fun
    docker:
    - image: circleci/node:10.16.0
    steps:
    - checkout
    - restore_cache:
        key: dependency-cache-{{ checksum "package-lock.json" }}
    - run:
        name: Install dependencies
        command: npm install
    - save_cache:
        key: dependency-cache-{{ checksum "package-lock.json" }}
        paths:
        - ./node_modules
    - run:
        name: Check lint errors
        command: npm run lint
    - run:
        name: Unit testing
        command: npm t
