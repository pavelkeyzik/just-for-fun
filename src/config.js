export const apiBase =
  process.env.NODE_ENV !== 'production'
    ? 'https://just-for-fun-server.herokuapp.com/'
    : 'http://localhost:3500/';

export const projectRepository = 'https://github.com/pavelkeyzik/just-for-fun';
export const projectName = 'JUSTFORFUN';
