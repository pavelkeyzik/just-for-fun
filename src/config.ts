export const isProduction = process.env.NODE_ENV === 'production';

export const apiBase = isProduction
  ? 'https://just-for-fun-server.herokuapp.com/'
  : 'http://localhost:3500/';

export const projectRepository = 'https://github.com/pavelkeyzik/just-for-fun';
export const projectName = 'JUSTFORFUN';
export const projectNameAccessible = 'Just for fun';

export const mapBoxAPIAccessToken = process.env.MAPBOX_ACCESS_TOKEN;

export const routes = {
  home: '/',
  places: '/places',
  contributors: '/contributors',
  admin: {
    home: '/admin',
    login: '/admin/login',
    places: '/admin/places',
    createNewPlace: '/admin/create-new-place',
    events: '/admin/events',
  },
};

export const defaultMarkerLat = 53.888019427799776;
export const defaultMarkerLng = 27.544588765497792;
