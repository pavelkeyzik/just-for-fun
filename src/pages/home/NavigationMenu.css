.NavigationMenu__container a {
  color: var(--link-foreground);
}

.NavigationMenu__container a:hover {
  color: var(--link-foreground-hover);
  text-decoration: underline;
}
