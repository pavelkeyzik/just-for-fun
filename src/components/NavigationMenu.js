import React from 'react';
import { Link } from 'react-router-dom';
import {
  Container,
  Navbar,
  NavbarBrand,
  Nav,
  NavItem,
  NavLink,
} from 'reactstrap';

import { projectRepository, projectName } from '../config';
import './NavigationMenu.css';

export function NavigationMenu() {
  return (
    <Container>
      <Navbar dark>
        <NavbarBrand tag={Link} to="/" aria-label="Home page">
          {projectName}
        </NavbarBrand>
        <Nav className="NavigationMenu__container">
          <NavItem>
            <NavLink href={projectRepository}>GitHub</NavLink>
          </NavItem>
        </Nav>
      </Navbar>
    </Container>
  );
}
