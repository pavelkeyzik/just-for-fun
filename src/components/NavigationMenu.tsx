import React from 'react';
import { Link } from 'react-router-dom';
import {
  Container,
  Navbar,
  NavbarBrand,
  Nav,
  NavItem,
  NavLink,
} from 'reactstrap';

import { projectRepository, projectName, routes } from '../config';
import styles from './NavigationMenu.module.css';

export function NavigationMenu(): JSX.Element {
  return (
    <Container>
      <Navbar dark>
        <NavbarBrand tag={Link} to={routes.home} aria-label="Home page">
          {projectName}
        </NavbarBrand>
        <Nav className={styles.menu}>
          <NavItem>
            <NavLink tag={Link} to={routes.contributors}>
              Contributors
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink href={projectRepository}>GitHub</NavLink>
          </NavItem>
        </Nav>
      </Navbar>
    </Container>
  );
}
