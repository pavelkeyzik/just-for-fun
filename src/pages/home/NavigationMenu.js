import React from 'react';
import {
  Container,
  Navbar,
  NavbarBrand,
  Nav,
  NavItem,
  NavLink,
} from 'reactstrap';

import { projectRepository, projectName } from '../../config';

export function NavigationMenu() {
  return (
    <Container>
      <Navbar dark>
        <NavbarBrand>{projectName}</NavbarBrand>
        <Nav navbar>
          <NavItem>
            <NavLink href={projectRepository}>GitHub</NavLink>
          </NavItem>
        </Nav>
      </Navbar>
    </Container>
  );
}
