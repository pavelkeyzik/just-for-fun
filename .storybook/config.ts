import { configure, addDecorator } from '@storybook/react';

import '../src/setup';

configure(require.context('../src', true, /\.stories\.tsx?$/), module);
