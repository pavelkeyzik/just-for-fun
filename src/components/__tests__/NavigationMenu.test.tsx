import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { NavigationMenu } from '../NavigationMenu';
import { projectName, projectRepository } from '../../config';

jest.mock('../../config', () => {
  return {
    projectName: 'PROJECT_NAME',
    projectRepository: 'PROJECT_LINK',
    routes: {
      home: '/',
      places: '/places',
      contributors: '/contributors',
    },
  };
});

test('Title should be equal to projectName field in config', () => {
  const { getByText } = render(
    <Router>
      <NavigationMenu />
    </Router>,
  );
  const title = getByText(projectName).textContent;

  expect(title).toBe(projectName);
});

test('Link to GitHub repo should be equal to projectRepository field in config', () => {
  const { getByText } = render(
    <Router>
      <NavigationMenu />
    </Router>,
  );
  const repoLinkElement = getByText('GitHub');

  expect(repoLinkElement.getAttribute('href')).toBe(projectRepository);
});
