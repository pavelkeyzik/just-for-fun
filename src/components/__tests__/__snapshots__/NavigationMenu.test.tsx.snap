// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check snapshot 1`] = `
<div
  class="container"
>
  <nav
    class="navbar navbar-dark"
  >
    <a
      aria-label="Home page"
      class="navbar-brand"
      href="/"
    >
      PROJECT_NAME
    </a>
    <ul
      class="menu nav"
    >
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="/contributors"
        >
          Contributors
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="PROJECT_LINK"
        >
          GitHub
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
